### PR TITLE
Gitea related fixes

### PR DESCRIPTION
**Pull request by deniscep**
- Adds an exception for UDP 123 - access to time sync for all components that need it
- Fixes errors in gitea repo migration scripts
- Modifies how keyVaultReferenceIdentity is set

### CODE REVIEW

**Review by tompaana — APPROVED** [review]

File: templates/shared_services/gitea/terraform/gitea-webapp.tf [hunk 1]
```diff
@@ -264,6 +264,6 @@ resource "azurerm_role_assignment" "gitea_acrpull_role" {
 # unfortunately we have to tell the webapp to use the user-assigned identity when accessing key-vault, no direct tf way.
 resource "null_resource" "webapp_vault_access_identity" {
   provisioner "local-exec" {
-    command = "az resource update --ids ${azurerm_app_service.gitea.id} --set properties.keyVaultReferenceIdentity=${azurerm_user_assigned_identity.gitea_id.id}"
+    command = "az rest --method PATCH --uri \"${azurerm_app_service.gitea.id}?api-version=2021-01-01\" --body \"{'properties':{'keyVaultReferenceIdentity':'${azurerm_user_assigned_identity.gitea_id.id}'}}\""
```
Comment (tamirkamara) [inline templates/shared_services/gitea/terraform/gitea-webapp.tf:267]: Doesn't the old way work?

File: scripts/gitea_migrate_repo.sh [hunk 2]
```diff
@@ -67,28 +67,28 @@ then
   credentials=$username:$password
   data='{"name": "'${username}'"}'
   url=${giteaUrl}/api/v1/users/${username}/tokens
-
+  
   # Create new access token
   response=$(curl -X POST -H "Content-Type: application/json" -k -d "${data}" -u ${credentials} ${url})
   token=$(jq -r '.sha1' <<< "$response")
-
+
```
[the remark this answers is not in the document]
Comment (Lybecker) [reply scripts/gitea_migrate_repo.sh:74]: You are having fun @tompaana :)

File: scripts/gitea_migrate_repo.sh [hunk 3]
```diff
@@ -67,28 +67,28 @@ then
...
   "lfs": false,
   "milestones": true,
   "mirror": true,
   "mirror_interval": "12h0m0s",
-  "private": true,
+  "private": false,
   "pull_requests": true,
   "releases": true,
-  "repo_name": "'${REPO_NAME}'",
+  "repo_name": "'${github_repo##*/}'",
```
Comment (Lybecker) [inline scripts/gitea_migrate_repo.sh:91]: Is the `##*/` supposed to be there?

Comment (deniscep) [reply scripts/gitea_migrate_repo.sh:89]: yes, that cuts the last / part of the URL and makes it into a name, so https://github.com/Microsoft/AzureTRE becomes AzureTRE #JustBashThings

**Pull request merged**